pyprland/plugins/scratchpads/__init__.py

```diff
@@ -747,6 +748,10 @@ async def run_hide(self, uid: str, force: bool = False, autohide: bool = False)
         for addr in scratch.extra_addr:
             await self.hyprctl(f"movetoworkspacesilent special:scratch_{uid},address:{addr}")
             await asyncio.sleep(0.01)
+
+        for e_uid in scratch.excluded_scratches:
+            await self.run_show(e_uid)
+            scratch.excluded_scratches.remove(e_uid)
```

Comment:
I think the iteration should force a copy of list, or, maybe preferably, empty the `excluded_scratches` list after the loop.